### PR TITLE
Add functional tests for plugins

### PR DESCRIPTION
**Pull request by sbs2001**
This test runs a simple http server in python which echoes back the request body. During the tests the http_plugin is enabled by modifying the profiles. It is configured to post the payload to the python server. Then the logs emitted by the server are asserted against expected results.

### CODE REVIEW

File: scripts/func_tests/config/config.yaml [hunk 1]
```diff
@@ -11,6 +11,8 @@ config_paths:
   simulation_path: /etc/crowdsec/simulation.yaml
   hub_dir: /etc/crowdsec/hub/
   index_path: /etc/crowdsec/hub/.index.json
+  notification_dir: /etc/crowdsec/notifications/
+  plugin_dir: /var/lib/crowdsec/plugins/
 crowdsec_service:
```
Comment (sabban) [inline scripts/func_tests/config/config.yaml:16]: This relies in `/usr/local/lib/crowdsec/plugins`

File: scripts/func_tests/config/config_no_agent.yaml [hunk 2]
```diff
@@ -12,6 +12,8 @@ config_paths:
   simulation_path: /etc/crowdsec/simulation.yaml
   hub_dir: /etc/crowdsec/hub/
   index_path: /etc/crowdsec/hub/.index.json
+  notification_dir: /etc/crowdsec/notifications/
+  plugin_dir: /var/lib/crowdsec/plugins/
 cscli:
```
Comment (sabban) [inline scripts/func_tests/config/config_no_agent.yaml:17]: same here

File: scripts/func_tests/config/config_no_capi.yaml [hunk 3]
```diff
@@ -11,6 +11,8 @@ config_paths:
   simulation_path: /etc/crowdsec/simulation.yaml
   hub_dir: /etc/crowdsec/hub/
   index_path: /etc/crowdsec/hub/.index.json
+  notification_dir: /etc/crowdsec/notifications/
+  plugin_dir: /var/lib/crowdsec/plugins/
 crowdsec_service:
```
Comment (sabban) [inline scripts/func_tests/config/config_no_capi.yaml:16]: still same here

File: scripts/func_tests/config/config_no_lapi.yaml [hunk 4]
```diff
@@ -11,6 +11,8 @@ config_paths:
   simulation_path: /etc/crowdsec/simulation.yaml
   hub_dir: /etc/crowdsec/hub/
   index_path: /etc/crowdsec/hub/.index.json
+  notification_dir: /etc/crowdsec/notifications/
+  plugin_dir: /var/lib/crowdsec/plugins/
 crowdsec_service:
```
Comment (sabban) [inline scripts/func_tests/config/config_no_lapi.yaml:16]: same again

File: scripts/func_tests/tests_post-install_7_plugin.sh [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    second_received_ip=$(cat mock_http_server_logs.log  | jq -r .request_body[1].decisions[0].value)
+    if [[ $second_received_ip != "1.2.3.5" ]] ; then
+        cleanup_tests
+        fail "expected to receive IP 1.2.3.5 as value of second decision"
+    fi 
+}
+
+setup_tests
+run_tests
+cleanup_tests
```
Comment (sabban) [inline scripts/func_tests/tests_post-install_7_plugin.sh:88]: Is there a way to shut the mock server ?

**Review by sabban — CHANGES_REQUESTED** [review]
few minor changes

File: scripts/func_tests/tests_post-install_7_plugin.sh (hunk 5 above)
Comment (sbs2001) [reply scripts/func_tests/tests_post-install_7_plugin.sh:88]: The `cleanup_tests` shuts the server. https://github.com/crowdsecurity/crowdsec/blob/3805cba6527a08a62b5494ed625318e2386e2db3/scripts/func_tests/tests_post-install_7_plugin.sh#L46

File: scripts/func_tests/config/config_no_lapi.yaml (hunk 4 above)
Comment (sbs2001) [reply scripts/func_tests/config/config_no_lapi.yaml:16]: fixed via https://github.com/crowdsecurity/crowdsec/pull/941/commits/4e34df049d71c0ab01816fe98a685c193a4afd66

File: scripts/func_tests/config/config_no_capi.yaml (hunk 3 above)
Comment (sbs2001) [reply scripts/func_tests/config/config_no_capi.yaml:16]: fixed via https://github.com/crowdsecurity/crowdsec/pull/941/commits/4e34df049d71c0ab01816fe98a685c193a4afd66

File: scripts/func_tests/config/config_no_agent.yaml (hunk 2 above)
Comment (sbs2001) [reply scripts/func_tests/config/config_no_agent.yaml:17]: fixed via https://github.com/crowdsecurity/crowdsec/pull/941/commits/4e34df049d71c0ab01816fe98a685c193a4afd66

**Review by sabban — APPROVED** [review]
lgtm

**Review by sabban — APPROVED** [review]
lgtm

**Pull request merged**